English version (#1)
yiochou/Resume: README.md

---

> ## Summary

Reviewer: 有好幾個 hands on experience 有點冗，覺得合併一起講比較好。
整段資訊並沒有很紮實，其實和最後的技術總結差別不大，只是把技術關鍵字用句子串起來，我覺得可以有一項是講比較 high level 你可以提供的能力、做到的事，比方說從零規劃打造一個微服務之類的

> - 3+ years of experience in backend development, including programing, database schema design, and performance tuning.

Reviewer: programming

> - Have solid experience in Node.js. Ability to write maintainable and readable code that follows SOLID principles.

Reviewer: 去掉 Have，都用名詞句子

> - The ability to break down requirements to minor tasks and implement these tasks on time.

Reviewer: ```suggestion
- The ability to break down requirements into minor tasks and complete them on time.
```

> - Hands-on experience of both relational database and NoSQL. Ability to design corresponding schema according to their features, and create reasonable indice.

Reviewer: Experience 接 in 或是 on
https://www.englishforums.com/English/ExperienceExperience/xcxrz/post.htm

> - Hands-on experience in building appliaction from scratch.

Reviewer: ```suggestion
- Hands-on experience in building application from scratch.
```

> - Challenge enjoying and being responsible at work.

Reviewer: 這句怪怪的 但不確定這樣改有符合你的意思嗎
Enjoy challenges and take full ownership for the results of my decisions

沒有用名詞句子，因為改成名詞句反而拗口

參考 https://enhancv.com/resume-skills/responsibility/

Author: 有捏 這樣好多了

---

## Work Experience

<!-- You have to wrap the "left" and "right" half of these headings in spans by
hand -->

### <span> SoundOn, Backend Engineer </span> 2020/7 - 2021/7

##### Tech: NodeJS, PostgresSQL, GCP, K8S, BigQuery, Dataflow, CircleCI, Firebase, React

> - Developed new features of podcast hosting and podcast player; furthermore, refactored the lagacy code base.

Reviewer: Legacy

Furthermore 不需要，這類沒有提供額外資訊的副詞能省即省

Podcast hosting 應該不是一個東西，可能講 podcast hosting service and player

> - Built alerting system on GCP, and integrated with Slack to notify developers actively if incident occrrs.

Reviewer: ```suggestion
- Built alert system on GCP, and integrated with Slack to notify developers in real-time.
- Built alert system on GCP, and integrated with Slack to notify developers whenever an incident occurs.
```
typo/spelling err 太多了啦

Author: 疑 所以不是 alerting system 嗎

Author: 疑 所以不是 alerting system 嗎
因為我有 google 到這個用寫法

Reviewer: 我是指 occurs (occrrs)

Author: 我內建的 typo detector 還沒升級

> - Scheduled ETL cron jobs to process data stored in BigQuery, and aggregated for the purpose of analytics and podcast ranking.

Reviewer: ```suggestion
- Scheduled ETL cron jobs to process and aggregate BigQuery data for performance analytics of podcast shows and podcast rankings.
```

Author: - data in BigQuery
- data stored in BigQuery
- BigQuery data

像這樣的情況 這三個用法上可以互通嗎

還有比如 podcast ranking / ranking of podcast shows 也可以互通嗎

Reviewer: 對 互通 偏好簡潔的 BigQuery data

> - Migrated database from Firestore (NoSQL) to PostgresSQL without downtime, and cleaned up dirty documents to consistent records.

Reviewer: ```suggestion
- Migrated database from Firestore (NoSQL) to PostgresSQL with zero downtime, and transformed dirty documents into consistent records.
```
nit: with zero downtime 個人偏好，聽起來比較有力
clean up A，不能 clean up A to B

> - Supported frontend team (using React, Redux, and configure Prerender), app team (using Flutter)。

Reviewer: ```suggestion
- Supported frontend team (React, Redux, Prerender), app team (Flutter)。
```

### <span> LJIT.io, Frontend/Backend Engineer </span> 2018/5 - 2020/2

##### Tech: NodeJS, Express.js, MongoDB, MySQL, Redis, AWS S3, React/Redux, Webpack, Docker

> - LJIT.io was an outsourcing company. Developed various projects and utilized multiple technology. Played both frontend and backend roles.

Reviewer: ```suggestion
- LJIT.io was an outsourcing company. Developed various projects and utilized multiple technologies. Played both frontend and backend roles.
```

> - LJIT.io was an outsourcing company. Developed various projects and touched multiple technology. At first, I was frontend developer, but then dedicated to backend development.

Reviewer: ```suggestion
- LJIT.io was an outsourcing company. Developed various projects and utilized multiple technology. Played both frontend and backend roles.
```

> - On blockchain game (dapp) project: Modified a smart contract and deployed to ethereum. Besides, developed website with React, Redux, and interacted with deployed smart contract.

Reviewer: ```suggestion
- Blockchain game (dapp) project: Modified a smart contract and deployed to ethereum. Developed website using React, Redux, and web3 to interact with the deployed smart contract.
```

> - On video streaming platform project: Developed both frontend and backend, and designed database schema. Stored video files to aws s3, and deployed CDN to improve viewer experience.

Reviewer: ```suggestion
- Video streaming platform project: Engineered frontend, backend, and database schema design. Stored video files on AWS S3, and deployed CDN for better viewer experience.
```
Developed 用太多次，抽換成 Engineered

可以參考一些 resume 常用動詞
https://www.themuse.com/advice/185-powerful-verbs-that-will-make-your-resume-awesome

> - On gambling platform project: Developed various algorithms of lotteries and wrote unit tests. Helped team spending less times on writing tests, by programming a script that converted test cases to unit tests code. Introduced two-factor authentication into login process.

Reviewer: ```suggestion
- Gambling platform project: Implemented various lottery algorithms and unit tests. Saved the team significant time on writing tests by introducing template for test code auto generation. Launched/Established two-factor authentication.
```
可以的話，加上具體省下多少時間

Author: 改的讚讚讚 (跪 

沒有具體的時間 (茶~

Reviewer: 我是要你 launched 和 established 選一個ＸＤ

- Took an active part in study group, and shared topics like SOLID, React/Redux, Database Transaction/Isolation and so on.

---

## Education

### <span> National Central University </span> 2013/9 - 2017/7

##### **Bachelor of Computer Science**.

---

## Skills

> - Programming Languages: Javascript(NodeJS, TypeScript), Golang

Reviewer: JavaScript
Node.js
